h/views/oidc.py

```diff
@@ -217,10 +218,17 @@ def connect_orcid_id(self, orcid_id: str, user: User | None):
 
     def log_in_with_orcid(self, orcid_id: str, user):
         if not user:
-            msg = "Not implemented yet"
-            raise RuntimeError(msg)
-
-        del orcid_id
+            # There's no Hypothesis account for this ORCID iD yet.
+            # Redirect to the "Sign up to Hypothesis with ORCID" page to create
+            # a new account.
+            return HTTPFound(
+                self._request.route_url(
+                    "signup.orcid",
+                    _query=encode_idinfo_token(
+                        self._jwt_service, orcid_id, JWTIssuers.OIDC_REDIRECT_ORCID
+                    ),
```

Comment:
Here the `/oidc/redirect/orcid` view is creating a JWT for the `/signup/orcid` view. This JWT and its format belong to the `/signup/orcid` view which consumes it. The `/oidc/redirect/orcid` is just one issuer of this JWT. The `/signup/orcid` view itself is another issue, and in future there could be more. So I think it makes sense for `views/oidc.py` to import a helper from `views/account_signup.py` in order to create this JWT: all the knowledge about this JWT belongs only in `account_signup.py`. This also achieves deduplication.